lua/swampshop/vgui/customizer.lua

```diff
@@ -296,523 +263,538 @@ local function Container(parent, label)
 
-        savebutton.DoClick = function(pnl)
-            if (IsValid(SS_CustTextureHistory)) then
-                SS_CustTextureHistory:Remove()
+            for x = 0, (LocalPlayer():GetBoneCount() - 1) do
+                local bn = LocalPlayer():GetBoneName(x)
+                local cleanname = cleanbonename(bn)
 
-                return
+                if cleanname ~= "__INVALIDBONE__" then
+                    choices[bn] = cleanname
+                end
             end
 
-            ImageHistoryPanel(pnl)
+            table.sort(choices, function(a, b) return a > b end)
+
+            SS_PreviewPane:AddChoiceProperty(choices, {initkey}, istable(v) and v or {}, (istable(v) and v.label or "Bone"))
         end
-    end
 
-    local rawzone = vgui.Create("DCollapsibleCategory", appearance_container)
-    rawzone:Dock(TOP)
-    rawzone:SetTall(256)
-    rawzone:DockMargin(0, 0, 0, 0)
-    rawzone:DockPadding(SS_COMMONMARGIN, SS_COMMONMARGIN, SS_COMMONMARGIN, SS_COMMONMARGIN)
-    rawzone.BasedPerformLayout = rawzone.PerformLayout
+        if (k == "submaterial") then
+            local choices = {}
 
-    rawzone.PerformLayout = function(pnl)
-        pnl:SizeToChildren(false, true)
-        pnl:InvalidateParent(true)
-        pnl:DockMargin(0, 0, appearance_container.VBar:IsVisible() and SS_COMMONMARGIN or 0, SS_COMMONMARGIN)
-        pnl:BasedPerformLayout(pnl:GetWide(), pnl:GetTall())
-    end
-
-    rawzone:SetLabel("Raw Data")
-    rawzone.Header:SetFont("SS_DESCINSTFONT")
+            for x = 0, math.min(31, (#(LocalPlayer():GetMaterials()) - 1)) do
+                local matname = LocalPlayer():GetMaterials()[x + 1] or ""
+                local exp = string.Explode("/", matname)
+                local nicename = exp[#exp]
+                choices[x] = tostring(x) .. " (" .. nicename .. ")"
+            end
 
-    rawzone.Header.UpdateColours = function(pnl)
-        pnl:SetTextColor(MenuTheme_TX)
-    end
+            SS_PreviewPane:AddChoiceProperty(choices, {initkey}, istable(v) and v or {}, (istable(v) and v.label or "Submaterial"))
+        end
 
-    rawzone.Header:SetContentAlignment(8)
-    rawzone.Header:SetTall(26)
-    rawzone.Paint = SS_PaintFG
-    rawzone:SetExpanded(false)
-    rawzone:SetKeyboardInputEnabled(true)
-    RAWENTRY_c = vgui.Create("DPanel", rawzone)
-    RAWENTRY_c:Dock(FILL)
-    RAWENTRY_c.Paint = SS_PaintBG
+        if (k == "color") then
+            self:AddColorOption(section_name, {initkey}, istable(v) and v.label or "Color")
+        end
 
-    RAWENTRY_c.PerformLayout = function(pnl)
-        pnl:SizeToChildren(false, true)
-        pnl:InvalidateParent(true)
+        if (k == "imgur") then
+            self:AddImgurOption(section_name, {initkey}, istable(v) and v.label or "Custom Texture")
+        end
     end
 
-    RAWENTRY = vgui.Create("DTextEntry", RAWENTRY_c)
+    local itmcw = self.item.configurable.wear
+  
+    self:AddSection("default")
+    print("sex")
```

Comment:
Please skim the code of your own PRs when you submit them to catch things like this